File: Pilha.java

public class Pilha<T>{
    T [] S;
    int top;

    public Pilha(int max){
        this.S = new T[max];
        this.top = -1;
    }

    boolean estaVazia(){
        return top==-1;
    }

    // PUSH
    void empilheirar(int x){
        top = top +1;
        S[top] = x;
    }

    // POP
    int desempilheirar() throws Exception{
        if(estaVazia()) throw new Exception(); 
        else{
            top = top -1;
            return S[top+1];
        }
    }


    public static void main(String[] args) {
        Pilha p = new Pilha(10);
        p.empilheirar(10);
        p.empilheirar(3);
        p.empilheirar(9);
        try {
            p.desempilheirar();
        } catch (Exception e) {
            e.printStackTrace();
        }
        p.empilheirar(30);
       System.out.println(); 
    }
}

File: pilha.py

class Pilha:

    def __init__(self, tam):
        self.S = [None] * tam
        self.topo = -1

    def esta_vazia(self):
        return self.topo == -1

    def esta_cheia(self):
        return self.topo == len(self.S) - 1

    def empilhar(self, novo):
        if not self.esta_cheia():
            self.topo = self.topo +1
            self.S[self.topo]  = novo
        else:
            print('Pilha cheia.')

    def desempilheirar(self):
        if not self.esta_vazia():
            self.topo = self.topo -1
            return self.S[self.topo +1]
        else:
            print('Pilha vázia')

    def imprimir_pilha(self):
        i = self.topo
        while ( i >= 0):
            print(self.S[i])
            i = i - 1                

p = Pilha(6)            
p.empilhar('Alvaro')
p.empilhar('David')
p.empilhar('Haniel')
p.empilhar('José')
p.empilhar('Luciano')
p.empilhar('Mariana')

p.desempilheirar()
p.desempilheirar()
p.desempilheirar()
p.desempilheirar()
p.desempilheirar()
p.desempilheirar()
p.desempilheirar()
p.imprimir_pilha()

